docs/build/tutorials/platform/subnets/create-a-evm-blockchain-on-subnet-with-avalanchejs.md

```diff
@@ -0,0 +1,545 @@
+
+All transactions on a blockchain need to be validated by the distributed nodes or validators. **Subnets** are the group of validators that can validate one or many chains. On Avalanche, there is one primary subnet, that currently validates 3 blockchains, and each chain serves a different purpose. One can easily deploy their subnets and invite other validators to be a part of it. A validator can be a part of multiple subnets, but it must be validating the primary subnet.
+
+On Avalanche, out of the 3 chains, **Platform Chain** (P-Chain) manages subnets, subnet validators, subnet blockchain, etc. In this tutorial, you will learn about creating your subnet and deploying **Ethereum Virtual Machine (EVM)** based blockchain on that subnet through your Node.js application using **AvalancheJS**. AvalancheJS is a javascript library that allows you to issue commands to the Avalanche node APIs without worrying about transaction serialization, signing it with the keys etc. On our [subnet series of tutorials](https://docs.avax.network/build/tutorials/platform/subnets/), you can get various great articles explaining subnets, building custom VMs, and a lot more.
+
+
+## Requirements
+
+* [Go](https://golang.org/doc/install) version >= 1.16.8
+* [NodeJS](https://nodejs.org/en) Node >= 10.16 and npm >= 5.6
+
+## Project structure
+
+Open the terminal and make a new directory at your desired location. We will keep all the binaries and other codes in this folder.
+
+```bash
+mkdir subnet-evm-demo
+cd subnet-evm-demo
+```
+
+## Setting up AvalancheGo and Subnet EVM Binaries
+
+Clients interact with Avalanche blockchain by issuing API calls to the nodes running `AvalancheGo`. We have to clone the repository, build its binary, and run it. If we want to deploy our blockchain, we have to put the blockchain's VM binary inside the `build/plugins` directory. Here we will also clone the `subnet-evm` repository, build the VM's binary, and copy it to the AvalancheGo's `build/plugins` directory. Follow the steps below -
+
+### Clone AvalancheGo Repository
+
+```bash
+git clone https://github.com/ava-labs/avalanchego
+cd avalanchego
+```
+
+### Build Binary
+
+Running the below command will create an `avalanchego` binary inside the `build/` directory and will install the coreth `evm` binary inside the `build/plugins` directory.
+
+```bash
+./scripts/build.sh
+```
+
+### Clone Subnet-EVM Repository
+
+Move to the `subnet-evm-demo` directory, and clone the repo.
+
+```bash
+git clone https://github.com/ava-labs/subnet-evm
+cd subnet-evm
+```
+
+### Build Binary and Copy it to AvalancheGo Plugins
+
+Now run the following command to build the VM's binary inside the `build/` directory, named as `srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy`. It is the id for this VM and corresponds to the string "**subnetevm**" zero-extended in a 32-byte array and encoded in CB58. Then copy it to AvalancheGo's `build/plugins` directory.
```

Comment:
Where is srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy from?

Reply:
It's a 32-byte array of `subnetevm` extended with zeros and encoded into CB58. I am not sure if there is any utility tool available for conversion, but explained a function [get32ByteCB58()](https://github.com/ava-labs/avalanche-docs/pull/635/files#diff-4091eb0b93faaacdaacd2b9c631048250c35f70e46cfd9fb93b4dfc10fa5f44fR388) later, which does the same.